breakwater/src/sinks/egui/dynamic_overlay.rs

```diff
@@ -107,11 +96,12 @@ pub fn load_and_check(dylib_path: impl AsRef<Path>) -> Result<UiOverlay, Error>
                 "dylib version ({dylib_versions:?}) do not match our version ({:?})",
                 breakwater_egui_overlay::VERSIONS
             );
-            return Err(VersionMismatchSnafu.into_error(snafu::NoneError));
+            return Err(eyre::eyre!("dynamic overlay version check failed"));
```

Comment:
FYI there is also `bail!()` exactly for that https://docs.rs/eyre/latest/eyre/macro.bail.html